iota-trinary/src/trytes.rs

```diff
@@ -0,0 +1,649 @@
+use alloc::string::String;
+use bytes::{BufMut, BytesMut};
+use core::ops::{self, Add, AddAssign, Deref, Index};
+use core::str::{self, FromStr};
+use core::{fmt, ptr};
+
+pub use core::convert::TryFrom;
+
+type Result<T> = core::result::Result<T, TrytesError>;
+
+#[derive(Debug)]
+pub enum TrytesError {
+    InvalidTryteStr,
+    InvalidTryteChar,
+    DecodeLength,
+}
+
+pub type Tryte = u8;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Trytes(BytesMut);
+
+impl Trytes {
+    /// Creates a new empty `Trytes`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let t = Trytes::new();
+    /// ```
+    pub fn new() -> Self {
+        Trytes(BytesMut::new())
+    }
+
+    /// Creates a new empty `Trytes` with a particular capacity.
+    ///
+    /// If the given capacity is `0`, no allocation will occur, and this method
+    /// is identical to the [`new`] method.
+    ///
+    /// [`new`]: #method.new
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::with_capacity(10);
+    ///
+    /// // The Trytes contains no chars, even though it has capacity for more
+    /// assert_eq!(t.len(), 0);
+    ///
+    /// // These are all done without reallocating...
+    /// let cap = t.capacity();
+    /// for _ in 0..10 {
+    ///     t.push('a');
```

Comment:
yeah, I should give `A` as example instead. But since Tryte RFC is on hold. This draft would probably not continue. I will close this draft, once 3T1B is going to implement.